### PR TITLE
New Dockerfile for Running Cypress in CI

### PR DESCRIPTION
**Pull request by manosnoam**
## Description
In order to run Cypress tests in CI (i.e. Jenkins) a designated Dockerfile was added to `/scripts/ci`.
It is based on Fedora 40 image (from Quay), and it uses NVM (Node Version Manager) to install Node v20.
It is a rootless image that is supported to be running within Openshift and Podman.

## How Has This Been Tested?
The Dockerfile has been built successfully both locally and on Quay.
To build with Podman run for example:
`podman build --pull --rm -f "scripts/ci/Dockerfile" -t odh-cypress:latest ./`

You can find recent image and build log in Quay:
https://quay.io/repository/nmanos/odh-dashboard?tab=builds

## Test Impact
Once the image build is available to download, for example:
`podman pull quay.io/nmanos/odh-dashboard:latest`

It can be consumed within CI framework that uses Containers for execution runtime.
It has been tested within Jenkins CI that uses Kubernetes Plugin to launch Jenkins agents.
For example, in Jenkins Pipeline, that runs inside a container of this image, it was tested as follow:
```
sh("npm --verbose --prefix frontend ci")
sh("npm --verbose run test:cypress-ci -- -s '**/application.cy.ts'")
```
The CI build of such test execution, produced the following report:
image

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

After the PR is posted & before it merges:
- [x] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: scripts/ci/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+COPY ./frontend/package*.json ./
+RUN chown $USER:0 ./*
+
+# Switch to the user, export env variables, and prepare NVM directories
+USER $USER
+ENV USER $USER
+ENV HOME $USER_HOME
+ENV NVM_DIR $USER_HOME/nvm
+ENV NODE_PATH "$NVM_DIR/versions/node/$NODE_VERSION/lib/node_modules"
+ENV PATH "$NPM_CACHE/bin:$NVM_DIR/versions/node/$NODE_VERSION/bin:/bin:$HOME/.local/bin:/root/.local/bin:$PATH"
```
Comment (jiridanek) [inline scripts/ci/Dockerfile:55]: afaik, people usually run things with `npx`, or why do you add NPM_CACHE to path?

Comment (manosnoam) [reply scripts/ci/Dockerfile]: Thanks, I removed `$NPM_CACHE/bin` from Path

**Review by FedeAlonso — APPROVED** [review]

**Pull request merged**